api/actions/manifest/state_collector_test.go

```diff
@@ -90,6 +97,63 @@ var _ = Describe("StateCollector", func() {
 		})
 	})
 
+	Describe("droplet", func() {
+		BeforeEach(func() {
+			appRepo.ListAppsReturns(repositories.ListResult[repositories.AppRecord]{Records: []repositories.AppRecord{{GUID: "app-guid", DropletGUID: "droplet-guid"}}}, nil)
+		})
+
+		It("gets the droplet", func() {
+			Expect(collectStateErr).NotTo(HaveOccurred())
+			Expect(dropletRepo.GetDropletCallCount()).To(Equal(1))
+			_, _, dropletGUID := dropletRepo.GetDropletArgsForCall(0)
+			Expect(dropletGUID).To(Equal("droplet-guid"))
+		})
+
+		It("returns empty droplet", func() {
```

Comment:
when the app has no droplet, the droplet repo would return a not found error. You should adjust the default droplet repo stubbing to reflect that.

Then the `Droplet` field on the `appState` should be either nil or zero, depending on whether it is a pointer or struct